helper/communicator/step_connect.go

```diff
@@ -53,6 +53,7 @@ func (s *StepConnect) Run(state multistep.StateBag) multistep.StepAction {
 			Config:      s.Config,
 			Host:        s.Host,
 			WinRMConfig: s.WinRMConfig,
+			WinRMPort:   s.SSHPort,
```

Comment:
The winrm_port option is to manually specify the port that the GUEST vm uses to listen to WinRM, this is fixing an issue where the HOST will try to connect to the GUEST WinRM port directly isntead of using the NAT created by virtualbox and potentially other builders.
Its the same implementation SSH uses to connect. The problem is that even when building with WinRM communicator, Virtualbox uses step_forward_ssh to create the port NAT wich stores the number under the variable referenced on this